lib/_http_client.js

```diff
@@ -858,9 +858,11 @@ function listenSocketTimeout(req) {
 }
 
 ClientRequest.prototype.onSocket = function onSocket(socket, err) {
-  // TODO(ronag): Between here and onSocketNT the socket
-  // has no 'error' handler.
-  process.nextTick(onSocketNT, this, socket, err);
+  if (err) {
+    process.nextTick(() => this.emit('error', err));
+  }
+
+  process.nextTick(onSocketNT, this, socket);
```

Comment:
There is quite a bit of logic in `onSocketNT`  that is not useless if the `err` argument is removed. I'm not sure if it is correct/wanted.

Reply:
All that is tearing down the socket. It happens if it's already marked as destroyed but it _also_ runs if it receives and error so I suspect that socket teardown actually still needs to be happen or this would be a resource leak. 🤔